### PR TITLE
add initial structure of the Service

### PR DESCRIPTION
**Pull request by mgoetzegb**
## What

Add the initial structure for the project. Current state with this PR:

- the service and its dependencies can the executed via docker compose
- the REST API is defined and swagger docs are served
- the database schema is defined
- health service is implemented
- database migration is implemented

No functionality regarding notifications is implemented and all rest endpoint merely return `HTTP 501 Not Implemented`.

Note: CI/CD will be refined after this PR, as it depends on several files added here.

## Why

This is a good starting point to implement the service.

## References

VTI-150

### CODE REVIEW

File: set_env_vars_local_setup.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# only intended to be used for the local setup
+
+# volues for which we don't have a default
```
Comment (endyApina) [inline set_env_vars_local_setup.sh:3]: typo in `value` for which.....

File: pkg/swagger/docsHandler.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// SPDX-FileCopyrightText: 2023 Greenbone AG <https://greenbone.net>
```
Comment (endyApina) [inline pkg/swagger/docsHandler.go:1]: Some of the license comments are `2024 Greenbone AG`
I guess this was an oversight.

File: pkg/repository/repository.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
+// Copyright (C) Greenbone Networks GmbH
```
Comment (endyApina) [inline pkg/repository/repository.go:1]:
```suggestion
// SPDX-FileCopyrightText: 2024 Greenbone AG <https://greenbone.net>
```

File: pkg/web/api.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
+// SPDX-FileCopyrightText: 2023 Greenbone AG <https://greenbone.net>
```
Comment (endyApina) [inline pkg/web/api.go:1]:
```suggestion
// SPDX-FileCopyrightText: 2024 Greenbone AG <https://greenbone.net>
```

Comment (endyApina) [reply pkg/web/api.go:1]: This suggested change is similar to others

File: cmd/notification-service/main.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
+package main
```
Comment (endyApina) [inline cmd/notification-service/main.go:1]:
```suggestion
// SPDX-FileCopyrightText: 2024 Greenbone AG <https://greenbone.net>
//
// SPDX-License-Identifier: AGPL-3.0-or-later

package main
```

File: pkg/swagger/docsHandler.go (hunk 2 above)
Comment (mgoetzegb) [reply pkg/swagger/docsHandler.go]: Yeah, hard to say. This file is from 2024, but a one to one copy from a file in another repo, which was created in 2024. Changed to 2024 though.

File: pkg/web/api.go (hunk 4 above)
Comment (mgoetzegb) [reply pkg/web/api.go]: Changed everywhere `2023 Greenbone AG` to ` 2024 Greenbone AG`

File: set_env_vars_local_setup.sh (hunk 1 above)
Comment (mgoetzegb) [reply set_env_vars_local_setup.sh]: thanks, fixed to `values`

File: README.md [hunk 6]
```diff
@@ -2,22 +2,75 @@
...
 
+## Usage
+
+The Notification Service is intended to be deployed along the others openSight services on the appliance. The service provides a REST API. See the [OpenApi definition](../api/notification-service/notificationService_swagger.yaml) for details. You can view them e.g. in the [Swagger Editor](https://editor.swagger.io/).
+
+Backend services can send notifications via the `Create Notification` endpoint. Those notifications can then be retrieved via `List Notifications` to provide them to the user.
+
 ## Requirements
 
+To run the service and its dependencies on you local machine you need a working installation of [docker](https://docs.docker.com/engine/install/) and `make`.
```
[the remark this answers is not in the document]
Comment (mgoetzegb) [reply README.md]: thanks, changed as suggested

**Review by endyApina — APPROVED** [review]

**Pull request merged**